app/services/registry-services.js

```diff
@@ -142,6 +141,85 @@ angular.module('registry-services', ['ngResource'])
+      //     },
+      //     {
+      //       "v1Compatibility": "{\"id\":\"5c5fb281b01ee091a0fffa5b4a4c7fb7d358e7fb7c49c263d6d7a4e35d199fd0\",\"created\":\"2015-12-08T18:31:50.979824705Z\",\"container\":\"ea7fe68f39fd0df314e841247fb940ddef4c02ab7b5edb0ee724adc3174bc8d9\",\"container_config\":{\"Hostname\":\"ea7fe68f39fd\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":null,\"Cmd\":[\"/bin/sh\",\"-c\",\"#(nop) ADD file:c295b0748bf05d4527f500b62ff269bfd0037f7515f1375d2ee474b830bad382 in /\"],\"Image\":\"\",\"Volumes\":null,\"WorkingDir\":\"\",\"Entrypoint\":null,\"OnBuild\":null,\"Labels\":null},\"docker_version\":\"1.8.3\",\"config\":{\"Hostname\":\"ea7fe68f39fd\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":null,\"Cmd\":null,\"Image\":\"\",\"Volumes\":null,\"WorkingDir\":\"\",\"Entrypoint\":null,\"OnBuild\":null,\"Labels\":null},\"architecture\":\"amd64\",\"os\":\"linux\",\"Size\":1113436}"
+      //     }
+      //   ],
+      // }
+      'query': {
+        method:'GET',
+        isArray: false,
+        transformResponse: function(data/*, headers*/){
+          var res = {};
+          var history = [];
+          var resp = angular.fromJson(data);
+          var v1Compatibility = undefined;
+
+          res.architecture = resp.architecture;
+
+          for (var idx in resp.history){
+            v1Compatibility = angular.fromJson(resp.history[idx].v1Compatibility);
+            
+            history.push({
+              id : v1Compatibility.id,
+              author : v1Compatibility.author,
+              os : v1Compatibility.os,
+              docker_version: v1Compatibility.docker_version,
+              created: v1Compatibility.created,
+              parent: v1Compatibility.parent,
+              labels: v1Compatibility.config.Labels
+            });
+          }
+          res.history = history;
+          res.os = history[0].os;
```

Comment:
Yes you are right. 

Concerning `v1Compatibility` its not very clear if it will stay or not. (see: [Docker Registry Api v2.0](https://docs.docker.com/v1.6/registry/spec/api/) & issues link at the top) 
Maybe I've missed the information, but anyway I agree that its implied that the API might be changed soon.

Concerning labels, do you mean its an exception because they are not always present? The author too might not be specified.